plugin.go

```diff
@@ -0,0 +1,283 @@
+	Bintray struct {
+		Repos []Repo `yaml:"repos"`
+	}
+
+	// BintrayConfig -> entrypoint for all configs
+	BintrayConfig struct {
+		Threads       int  `yaml:"threads"`
+		Cleanup       bool `yaml:"cleanup"`
+		Bintray       `yaml:"bintray"`
+		Username      string `yaml:"username"`
+		APIKey        string `yaml:"api_key"`
+		GPGPassphrase string `yaml:"gpg_passphrase"`
+	}
+
+	Plugin struct {
+		BintrayConfig      `yaml:"bintray_config"`
+		ServicesManager    *bintray.ServicesManager
+		BintrayConfigPath  string   `yaml:"bintray_cfg_path"`
+		Version            string   `yaml:"verison"`
+		UploadPackage      bool     `yaml:"uploadpackage"`
+		SignPackageVersion bool     `yaml:"signpackageversion"`
+		PublishPackage     bool     `yaml:"publishpackage"`
+		CalcMetadata       bool     `yaml:"calcmetadata"`
+		ShowPackage        bool     `yaml:"showpackage"`
+		Envs               []string `yaml:"envs"`
+	}
+)
+
+const defaultLicense = "Apache 2.0"
+
+func (p Plugin) Exec() error {
+	// var (
+	// 	files []string
+	// )
+	// fmt.Printf("Drone Bintray Plugin built from %s\n", p.Version)
+	cSig := make(chan os.Signal)
+	signal.Notify(cSig, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-cSig
+		cleanup()
+		os.Exit(1)
+	}()
+
+	if p.BintrayConfigPath != "" {
+		p.BintrayConfig.ReadConfig(p.BintrayConfigPath)
+	}
+
+	p.ServicesManager = p.BintrayConfig.InitConfig()
+
+	if p.BintrayConfig.Cleanup {
+		p.BintrayConfig.cleanup(p.ServicesManager)
+		p.BintrayConfig.checkDetails(p.ServicesManager)
+	}
+
+	for _, repo := range p.BintrayConfig.Bintray.Repos {
+		for _, pack := range repo.Packages {
+			uploaded := 0
+			if p.UploadPackage {
+				uploaded, _, _ = p.BintrayConfig.uploadPackage(p.ServicesManager, &repo, &pack)
+			}
+			if uploaded > 0 || p.SignPackageVersion {
+				p.BintrayConfig.signPackageVersion(p.ServicesManager, &repo, &pack)
+			}
+			if uploaded > 0 && pack.Upload.Publish {
+				p.BintrayConfig.publishPackage(p.ServicesManager, &repo, &pack)
+			}
+			if uploaded > 0 || p.CalcMetadata {
+				p.BintrayConfig.calcMetadata(p.ServicesManager, &repo, &pack)
+			}
+			if uploaded > 0 || p.ShowPackage {
+				p.BintrayConfig.showPackage(p.ServicesManager, &repo, &pack)
+			}
+		}
+	}
+	return nil
+}
+
+// InitConfig for initializing service manager for bintray functionalities
+func (config *BintrayConfig) InitConfig() *bintray.ServicesManager {
+
+	btDetails := auth.NewBintrayDetails()
+	btDetails.SetUser(config.Username)
+	btDetails.SetKey(config.APIKey)
+	btDetails.SetDefPackageLicense(defaultLicense)
+
+	serviceConfig := bintray.NewConfigBuilder().
+		SetBintrayDetails(btDetails).
+		SetDryRun(false).
+		SetThreads(config.Threads).
+		Build()
+
+	btManager, err := bintray.New(serviceConfig)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err.Error())
+	}
+
+	return btManager
+}
+
+func (config *BintrayConfig) deleteRepo(btManager *bintray.ServicesManager, repo *Repo) error {
+	fmt.Printf("\nDeleting Repo.. [%s]\n", repo.Name)
```

Comment:
With all these log messages you should use a structured logger like logrus, we will anyway use it more within plugins as it's part of our upcoming plugin lib

Reply:
right for some reason logger built in jfrog go library of bintray wasn't working. I was hoping to reuse that tho